python/ray/data/tests/test_dataset_iterator.py

```diff
@@ -52,6 +54,17 @@ def test_torch_conversion(ray_start_regular_shared):
         assert batch["value"].tolist() == list(range(5))
 
 
```

Comment:
Maybe also add a test that `iter_torch_batches` raises an error if `collate_fn` is specified with `dtype` or `device`?

Reply:
added test